source/tutorials/configure-selinux/configure-selinux.rst

```diff
@@ -0,0 +1,139 @@
+
+* In **enforcing** mode, it not only prints out the AVC errors, but also blocks the operations that violate the sepolicy rules.
+
+It's a good practice to set SELinux in permissive mode at the beginning of the development phase, so that you can easily bring up the system and fix any SELinux errors. After resolving all AVC errors, you can change the SELinux mode to enforcing mode.
+
+SELinux Configuration
+=====================
+
+To change SELinux in build time
+-------------------------------
+
+    The SELinux operation mode is configured using *mixins*. You can edit the '*mixins.spec*' file in the source folder for your lunch target at '*device/intel/project-celadon/<target-name>/mixins.spec*', and rebuilt the image to take effect. Set the '*sepolicy*' key to either **permissive** or **enforcing** to enable the corresponding SELinux mode.
+
+    .. code-block:: none
+        :emphasize-lines: 3
+
+        ...
+        project-celadon: default
+        sepolicy: permissive
+        graphics: project-celadon(gen9+=true,hwc2=true,vulkan=false,drmhwc=false,minigbm
+        ...
+
+    .. note::
+        1. The *permissive* mode only can be activated in **userdebug** or **eng** builds, as the user builds always operate in *enforcing* SELinux mode in Android.
+        2. After you change the *sepolicy* value in *mixins.spec*, you need to run the '*device/intel/mixins/mixin-update*' script to make the changes taking effect before starting the build.
+
+To change SELinux mode at runtime
+---------------------------------
+
+    Besides changing the SELinux operaion mode statically by editting the configuration file during the build time, users can also change the SELinux mode at runtime through *adb shell* commands.
+
+    * Get the current SELinux operation mode with ``getenforce`` command
+
+    .. code-block:: bash
+        :emphasize-lines: 2
+
+        $ adb shell
+        celadon:/ $ getenforce
+        Permissive
+
+
+    * Set the SELinux operation mode with ``setenforce`` command
+
+    .. code-block:: bash
+        :emphasize-lines: 3, 7
+
+        $ adb shell
+        celadon:/ $ su
+        celadon:/ # setenforce 0
+        celadon:/ # getenforce
+        Permissive
+
+        celadon:/ # setenforce 1
+        celadon:/ # getenforce
+        Enforcing
+
+    .. note::
+        1. You must run the ``setenforce`` command with *root* permission.
+        2. Changing the SELinux operation mode will return to its default mode specified in the *mixins.spec* file after system reboots.
+
+Add sepolicy rules for a module
+===============================
+
+When adding a new module to |C|, you may need to add relevant sepolicy rules, otherwise its operations may be blocked by SELinux. The following instructions take the '*rfkill*' module as an example, and show you the required steps to get accesses to the system resources.
```

Comment:
Suggested rewrite: 
When you add a new module to |C|, you might need to add relevant sepolicy rules, otherwise its operations might be blocked by SELinux. The following instructions take the '*rfkill*' module as an example, and show you the required steps to get access to the system resources.